stripe/src/main/java/com/stripe/android/paymentsheet/repositories/PaymentMethodsRepository.kt

```diff
@@ -15,22 +17,38 @@ internal sealed class PaymentMethodsRepository {
         type: PaymentMethod.Type
     ): List<PaymentMethod>
 
+    abstract suspend fun save(
+        customerConfig: PaymentSheet.CustomerConfiguration,
+        paymentMethodCreateParams: PaymentMethodCreateParams,
+    ): PaymentMethod
+
     /**
      * A [PaymentMethodsRepository] that uses a pre-determined list of payment methods.
      */
     class Static(
         private val paymentMethods: List<PaymentMethod>
     ) : PaymentMethodsRepository() {
+        var error: Throwable? = null
         override suspend fun get(
             customerConfig: PaymentSheet.CustomerConfiguration,
             type: PaymentMethod.Type
         ): List<PaymentMethod> = paymentMethods
+
+        override suspend fun save(
+            customerConfig: PaymentSheet.CustomerConfiguration,
+            paymentMethodCreateParams: PaymentMethodCreateParams
+        ): PaymentMethod {
```

Comment:
We should probably drop the `Static` implementation and make `PaymentMethodsRepository` an interface. We only need this logic in tests.